applications/MeshingApplication/custom_utilities/mmg_utilities.h

```diff
@@ -0,0 +1,645 @@
+ * @details This class are different utilities that uses the MMG library
+ * @author Vicente Mataix Ferrandiz
+ */
+template<MMGLibray TMMGLibray>
+class KRATOS_API(MESHING_APPLICATION) MmgUtilities
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of MmgUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(MmgUtilities);
+
+    /// Node definition
+    typedef Node <3>                                                   NodeType;
+    // Geometry definition
+    typedef Geometry<NodeType>                                     GeometryType;
+
+    /// Conditions array size
+    static constexpr SizeType Dimension = (TMMGLibray == MMGLibray::MMG2D) ? 2 : 3;
+
+    /// Conditions array size
+    static constexpr SizeType ConditionsArraySize = (Dimension == 2) ? 1 : 2;
+
+    /// Elements array size
+    static constexpr SizeType ElementsArraySize = (Dimension == 2) ? 1 : 2;
+
+    /// The type of array considered for the tensor
+    typedef typename std::conditional<Dimension == 2, array_1d<double, 3>, array_1d<double, 6>>::type TensorArrayType;
+
+    /// Double vector
+    typedef std::vector<double> DoubleVectorType;
+
+    /// Double vector map
+    typedef std::unordered_map<DoubleVectorType, IndexType, KeyHasherRange<DoubleVectorType>, KeyComparorRange<DoubleVectorType> > DoubleVectorMapType;
+
+    /// Index vector map
+    typedef std::unordered_map<IndexVectorType, IndexType, KeyHasherRange<IndexVectorType>, KeyComparorRange<IndexVectorType> > IndexVectorMapType;
+
+    ///@}
+    ///@name  Enum's
+    ///@{
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief It prints info about the current mesh
+     * @param rNumberOfNodes The number of nodes
+     * @param rNumberOfConditions The number of conditions
+     * @param rNumberOfElements The number of elements
+     * @param EchoLevel The level of verbosity
+     */
+    void PrintMmgMeshInfo(
+        SizeType& rNumberOfNodes,
+        array_1d<SizeType, 2>& rNumberOfConditions,
+        array_1d<SizeType, 2>& rNumberOfElements,
+        const IndexType EchoLevel = 0
+        );
+
+    /**
+     * @brief It checks if the nodes are repeated and remove the repeated ones
+     * @param rModelPart The model part of interest to study
+     * @param EchoLevel The level of verbosity
+     */
+    IndexVectorType CheckNodes(
+        ModelPart& rModelPart,
+        const IndexType EchoLevel = 0
+        );
+
+    /**
+     * @brief It checks if the conditions are repeated and remove the repeated ones
+     * @param EchoLevel The level of verbosity
+     */
+    IndexVectorType CheckConditions0(const IndexType EchoLevel = 0);
```

Comment:
Function names and comments in the header file make it impossible to determine what this does without digging into the code. The comments says it removes the repeated conditions, but looking at the code, it only returns the vector of indices to be removed. This could already be expressed in the function name without the comment.

Reply:
OK

Reply:
Thanks for improving the function names. However you only fixed the comment in one place. It has copies in other places and the copies are still wrong. Please fix or remove the copies.